exercises/rna-transcription/description.md

```diff
@@ -15,3 +15,7 @@ each nucleotide with its complement:
 * `C` -> `G`
 * `T` -> `A`
 * `A` -> `U`
+
+Note that since we are finding the complement, you should use the mapping of pairs above (and not just swap T for U).
+
+Your function will need to be able to handle invalid inputs, both partial (a few invalid bases) and completely invalid strings. In these cases it should return an empty string.
```

Comment:
> In these cases it should return an empty string.

No! No! No!

Returning the empty string is a valid complement for the empty sequence as input.

Yes, it is part of this exercise to handle invalid input, because we use a string as input which has a much larger domain than just Nucleotides.

We have choosen to do this even on languages that could completely avoid invalid sequences because of their typesystem, because we treat it as a parsing and string transformation exercise.

But please, lets use the idiomatic error mechanisms of the implementing language. Lets throw something in Java, raise in Ruby, return an error-tuple/atom in elixir, erlang and LFE, lets have a `Result`- or `Either`-, or `Option`-, or `Maybe`-type in Rust, OCaml, Haskell, Idris, return an error-value in Go, do whatever is idiomatic in Python, but please do not **force** to use something unidiomatic via the description.

Of course tracks can choose to simply expect the empty string to signal errors, because they do not want to introduce proper error handling in this exercise but it should be the tracks maintainers who choose, not the description. Error handling is a subject that has to be treatened carefully…

Reply:
Right now, the empty string is what passes the test in Python though, which is how I found this issue (I had no idea we even had to handle invalid cases).

Perhaps this should be changed there, and the descriptions changed in both to reflect that.

Reply:
Please PR against their `HINTS.md` in that exercise with an accompanied regenerated README.md to describe the error handling part, but leave it here only as “There may be errors in the input which you need to detect and act accordingly” or something like that.

Reply:
@NobbZ, you're right regarding the error handling - I've submitted a PR to fix the Python implementation.